paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/kafka/formats/ogg/OggRecordParser.java

```diff
@@ -0,0 +1,218 @@
+                break;
+            case OP_INSERT:
+                checkNotNull(root.get(FIELD_AFTER), errorMessageTemplate, FIELD_AFTER);
+                break;
+            case OP_DELETE:
+                checkNotNull(root.get(FIELD_BEFORE), errorMessageTemplate, FIELD_BEFORE);
+                checkNotNull(root.get(FIELD_AFTER), errorMessageTemplate, "null");
+                break;
+        }
+    }
+
+    @Override
+    protected String extractString(String key) {
+        if (key.equals(FIELD_TABLE)) {
+            extractDatabaseAndTableNames();
+            return tableName;
+        } else if (key.equals(FIELD_DATABASE)) {
+            extractDatabaseAndTableNames();
+            return databaseName;
+        }
+        return root.get(key) != null ? root.get(key).asText() : null;
+    }
+
+    private void extractDatabaseAndTableNames() {
+        JsonNode tableNode = root.get(FIELD_TABLE);
+        if (tableNode != null) {
+            String[] dbt = tableNode.asText().split("\\.", 2); // Limit split to 2 parts
+            if (dbt.length == 2) {
+                databaseName = dbt[0];
+                tableName = dbt[1];
+            }
+        }
+    }
+
+    private LinkedHashMap<String, DataType> extractFieldTypesFromOracleType() {
+        LinkedHashMap<String, DataType> fieldTypes = new LinkedHashMap<>();
+
+        JsonNode record = root.get(FIELD_AFTER);
+        Map<String, Object> linkedHashMap =
+                OBJECT_MAPPER.convertValue(
+                        record, new TypeReference<LinkedHashMap<String, Object>>() {});
+        if (linkedHashMap == null) {
+            return new LinkedHashMap<>();
+        }
+
+        Set<String> keySet = linkedHashMap.keySet();
+        String[] columns = keySet.toArray(new String[0]);
+        for (String column : columns) {
+            fieldTypes.put(toFieldName(column), DataTypes.STRING());
+        }
+        return fieldTypes;
+    }
+
+    private Map<String, String> extractRow(
+            JsonNode record, LinkedHashMap<String, DataType> paimonFieldTypes) {
+        Map<String, String> linkedHashMap =
+                OBJECT_MAPPER.convertValue(
+                        record, new TypeReference<LinkedHashMap<String, String>>() {});
+        if (linkedHashMap == null) {
+            return new HashMap<>();
+        }
+
+        Map<String, String> resultMap = new HashMap<>();
+        for (Map.Entry<String, String> entry : linkedHashMap.entrySet()) {
+            String key = entry.getKey();
+            String value = entry.getValue();
+            paimonFieldTypes.put(key, DataTypes.STRING());
+            resultMap.put(key, value);
+        }
+
+        // generate values for computed columns
+        for (ComputedColumn computedColumn : computedColumns) {
+            resultMap.put(
+                    computedColumn.columnName(),
+                    computedColumn.eval(resultMap.get(computedColumn.fieldReference())));
```

Comment:
Computed column value is evaluated, should pad the paimonFieldTypes?
I also noticed that CanalRecordParser `extractRecords` has the same problem, please fix it.

Reply:
done.